tests/components/onewire/test_config_flow.py

```diff
@@ -1,26 +1,54 @@
 """Tests for 1-Wire config flow."""
+from collections.abc import Generator
 from unittest.mock import AsyncMock, patch
 
 from pyownet import protocol
 import pytest
 
-from homeassistant.components.onewire.const import DOMAIN
+from homeassistant.components.onewire.const import (
+    DOMAIN,
+    INPUT_ENTRY_CLEAR_OPTIONS,
+    INPUT_ENTRY_DEVICE_SELECTION,
+)
 from homeassistant.config_entries import SOURCE_USER, ConfigEntry
 from homeassistant.const import CONF_HOST, CONF_PORT
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.config_validation import ensure_list
+
+from .const import MOCK_OWPROXY_DEVICES
 
 
 @pytest.fixture(autouse=True, name="mock_setup_entry")
-def override_async_setup_entry() -> AsyncMock:
+def override_async_setup_entry() -> Generator[AsyncMock, None, None]:
     """Override async_setup_entry."""
     with patch(
         "homeassistant.components.onewire.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
 
 
-async def test_user_flow(hass: HomeAssistant, mock_setup_entry: AsyncMock):
+@pytest.fixture
+def filled_device_registry(
```

Comment:
We're using the async api so we should make sure the fixture runs inside the event loop.
```suggestion
async def filled_device_registry(
```